### ISSUE TITLE
Logs by different containers are hard to distinguish

### ISSUE THREAD

**Issue by hannes-ucsc** [DataBiosphere/azul#4468]
… since they're all prefixed with `docker`. The prefix should be the container name.

```
$ grep -iC3 'docker: clam' /var/log/messages
Feb 26 08:00:05 ip-172-71-0-215 docker: ==> /var/log/gitlab/nginx/gitlab_access.log <==
Feb 26 08:00:05 ip-172-71-0-215 docker: 172.71.3.154 - - [26/Feb/2023:08:00:05 +0000] "POST /api/v4/jobs/request HTTP/1.1" 204 0 "" "gitlab-runner 15.8.0 (15-8-stable; go1.18.9; linux/amd64)" -
Feb 26 08:00:05 ip-172-71-0-215 docker: Checking for jobs...nothing                       #033[0;m  runner#033[0;m=nJ9KdGyH
Feb 26 08:00:05 ip-172-71-0-215 docker: ClamAV update process started at Sun Feb 26 08:00:05 2023
Feb 26 08:00:05 ip-172-71-0-215 docker: daily database available for update (local version: 26816, remote version: 26823)
Feb 26 08:00:05 ip-172-71-0-215 docker: ==> /var/log/gitlab/sshd/current <==
Feb 26 08:00:05 ip-172-71-0-215 docker: 2023-02-26_08:00:05.94328 kex_exchange_identification: Connection closed by remote host
--
Feb 26 13:36:13 ip-172-71-0-215 docker: Time: 20153.704 sec (335 m 53 s)
Feb 26 13:36:13 ip-172-71-0-215 docker: Start Date: 2023:02:26 08:00:19
Feb 26 13:36:13 ip-172-71-0-215 docker: End Date:   2023:02:26 13:36:13
Feb 26 13:36:13 ip-172-71-0-215 docker: clamscan succeeded
Feb 26 13:36:14 ip-172-71-0-215 dockerd: time="2023-02-26T13:36:14.109945217Z" level=info msg="ignoring event" container=37d154a8c34ced028d1fcbb2c911e20b3e082bdc784d0186d60650de049fb0b6 module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
Feb 26 13:36:14 ip-172-71-0-215 containerd: time="2023-02-26T13:36:14.131778842Z" level=info msg="shim disconnected" id=37d154a8c34ced028d1fcbb2c911e20b3e082bdc784d0186d60650de049fb0b6
Feb 26 13:36:14 ip-172-71-0-215 containerd: time="2023-02-26T13:36:14.138505989Z" level=warning msg="cleaning up after shim disconnected" id=37d154a8c34ced028d1fcbb2c911e20b3e082bdc784d0186d60650de049fb0b6 namespace=moby
```

Note the `docker: ClamAV update process started` line and the preceding `docker: 172.71.3.154 … POST …` line. The former is from `freshclam` inside the `clamav` container, the latter is from `nginx` inside the `gitlab` container.

**Comment by hannes-ucsc** [comment]
This lacks a reproduction. I think we came across this when demoing the ClamAV scanning of the GitLab instance (#3614).

**Comment by hannes-ucsc** [comment]
Repro added.

**Comment by hannes-ucsc** [comment]
The blocker #3894 will remove the two-line log entries from the `gitlab` container, like this one:

```
Feb 26 08:00:05 ip-172-71-0-215 docker: ==> /var/log/gitlab/nginx/gitlab_access.log <==
Feb 26 08:00:05 ip-172-71-0-215 docker: 172.71.3.154 - - [26/Feb/2023:08:00:05 +0000] "POST /api/v4/jobs/request HTTP/1.1" 204 0 "" "gitlab-runner 15.8.0 (15-8-stable; go1.18.9; linux/amd64)" -
```

We only need to deal with vanilla container output.

[Configure](https://docs.docker.com/config/containers/logging/syslog/) syslog in `/etc/docker/daemon.json` to format the syslog lines differently by default, equivalent to what this command line does for a single container:

```
$ docker run --log-driver syslog --log-opt tag="docker: {{.ImageName}}/{{.Name}}/{{.ID}}" alpine echo hello world
hello world
# grep hello /var/log/messages
Feb 28 21:59:00 ip-172-71-0-215 docker: alpine/naughty_haslett/196e804ca9d6[4168]: hello world
```

**Comment by hannes-ucsc** [comment]
Interestingly, the blockee is already in progress. This might have been fixed by #3894. Assignee to check the repro again, as a spike. If not-reproducible, close ticket and reset estimate to zero.

**Comment by hannes-ucsc** [comment]
For demo, using CloudWatch Logs Insights, show that log messages from containers are prefixed with the container name and that no log messages are duplicated.
